docs/source/developers/guide/resources.rst

```diff
@@ -31,9 +31,10 @@ Additional information and resources
 On this page we have listed resources that may be relevant or useful for
 contributors who want to learn more about different parts of Apache Arrow.
 
-.. Annotation Glossary
-.. -------------------
-.. ARROW-15130
+Annotation Glossary
```

Comment:
just a drive-by nit: what is the 'annotation' part here referring to?

Reply:
To the short description of the term in the Glossary. It is mainly left from when searching for the projects having the glossary and saw they are naming it Annotation Glossary. Will remove it so it is in sync with [ARROW-15130](https://issues.apache.org/jira/browse/ARROW-15130) 👍